sdk/anomalydetector/ai-anomaly-detector-rest/samples/v1-beta/typescript/src/sample_multivariate_detection.ts

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * Demonstrates how to train a model on multivariate data and use this model to detect anomalies.
+ *
+ * @summary detect multivaariate anomalies.
+ */
+
+import createAnomalyDetectorRestClient, {
+  BatchDetectAnomalyParameters,
+  CreateMultivariateModelParameters,
+  DetectionResultOutput,
+  ListMultivariateModelParameters,
+  ModelOutput,
+  paginate,
+} from "@azure-rest/ai-anomaly-detector";
+import { AzureKeyCredential } from "@azure/core-auth";
+
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+dotenv.config();
+
+// You will need to set this environment variables or edit the following values
+const apiKey = process.env["ANOMALY_DETECTOR_API_KEY"] || "";
+const endpoint = process.env["ANOMALY_DETECTOR_ENDPOINT"] || "";
+const apiVersion = "v1.1";
+
+const dataSource = "<your data source>";
+
+function sleep(time: number): Promise<NodeJS.Timer> {
+  return new Promise((resolve) => setTimeout(resolve, time));
+}
+
+export async function main() {
+  // create client
+  const credential = new AzureKeyCredential(apiKey);
+  const client = createAnomalyDetectorRestClient(endpoint, apiVersion, credential);
+
+  // Already available models
+  const options: ListMultivariateModelParameters = {
+    queryParameters: { skip: 0, top: 10 },
+  };
+  const initialResponse = await client.path("/multivariate/models").get(options);
+  const pageData = paginate(client, initialResponse);
+  const listModelsResult = [];
+  for await (const item of pageData) {
+    listModelsResult.push(item);
+  }
+  console.log(listModelsResult);
+
+  // construct model request (notice that the start and end time are local time and may not align with your data source)
+  const createMultivariateModelParameters: CreateMultivariateModelParameters = {
+    body: {
+      alignPolicy: {
+        alignMode: "Outer",
+        fillNAMethod: "Linear",
+        paddingValue: 0,
+      },
+      dataSchema: "MultiTable",
+      dataSource: dataSource,
+      displayName: "Devops-MultiAD",
+      endTime: "2021-01-02T05:00:00Z",
+      slidingWindow: 200,
+      startTime: "2021-01-02T00:00:00Z",
+    },
+    headers: { "Content-Type": "application/json" },
+  };
+
+  // train model
+  const createModelResult = await client
+    .path("/multivariate/models")
+    .post(createMultivariateModelParameters)
+    .then((res) => {
+      if (!("modelId" in res.body)) {
+        throw res.body;
+      }
+      return res.body;
+    });
+  console.log(createModelResult);
+
+  // get model status
+  const modelId = createModelResult.modelId;
+  let modelResponse = await client
+    .path("/multivariate/models/{modelId}", modelId)
+    .get()
+    .then((res) => {
+      if (!("modelInfo" in res.body)) {
```

Comment:
It seems that the helper function isUnexpected is not working fine. 
isUnexpected(modelResponse) is always return true (without ".then" part). 

I think it is caused by the responseMap is not generated correctly. 
responseMap in source code:
`
const responseMap: Record<string, string[]> = {
   "GET /multivariate/models/{modelId}": ["200"],
...
};
`
actual input
`
GET /anomalydetector/v1.1/multivariate/models
`

Reply:
@MaryGao this is another example of `isUnexpected` having issues with parametrized host.